ingester/src/persist/worker.rs

```diff
@@ -321,6 +323,44 @@ where
     (catalog_sort_key_update, parquet_table_data)
 }
 
+/// Fetch the table column map from the catalog and verify if they contain all columns in the sort key
+async fn fetch_column_map<O>(
+    ctx: &Context,
+    worker_state: &SharedWorkerState<O>,
+    // NOTE: CALLER MUST LOAD SORT KEY BEFORE CALLING THIS FUNCTION EVEN IF THE sort key IS NONE
+    // THIS IS A MUST TO GUARANTEE THE RETURNED COLUMN MAP CONTAINS ALL COLUMNS IN THE SORT KEY
+    // The purpose to put the sort_key as a param here is to make sure the caller has already loaded the sort key
+    sort_key: Option<SortKey>,
```

Comment:
This function doesn't need ownership of the sort key; this can be changed to:

```suggestion
    sort_key: Option<&SortKey>,
```

and then line 181 can be:

```
let columns = fetch_column_map(ctx, worker_state, sort_key.as_ref()).await?;
```

to save an allocation.